### PR TITLE
Polishing README.

### PR DESCRIPTION
**Pull request by concretevitamin**
Main change is to promote "unified interface" a bit.

All: please help look over to see if the new messaging is better.

- [x] port final changes to docs
- [ ] optimizer; provision

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,12 +12,12 @@ SkyPilot is a framework for easily running machine learning workloads[^1] on any
...
 Use the clouds **easily** and **cost effectively**, without needing cloud infra expertise.
 
-_Ease of use & productivity_
+_Ease of use_
 * **Run existing projects on the cloud** with zero code changes
-* **Easily manage jobs** across multiple clusters
-* **Automatic fail-over** to find scarce resources (GPUs) across regions and clouds
-* **Store datasets on the cloud** and access them like you would on a local file system 
-* **No cloud lock-in** – seamlessly run your code across different cloud providers (AWS, Azure or GCP)
+* Use a **unified interface** to run on any cloud, without vendor lock-in (currently AWS, Azure, GCP)
```
Comment (romilbhardwaj) [inline README.md:17]: nit: remove `Use a ` to start directly with the main keyword. e.g. `**Unified interface** to run jobs on any cloud, without vendor lock-in (currently AWS, Azure, GCP)`

File: README.md [hunk 2]
```diff
@@ -12,12 +12,12 @@ SkyPilot is a framework for easily running machine learning workloads[^1] on any
...
+* Easily **queue jobs** on one or multiple clusters
```
Comment (romilbhardwaj) [inline README.md:18]: Can remove easily `**Queue jobs** on one or multiple clusters`

File: README.md [hunk 3]
```diff
@@ -33,9 +33,9 @@ You can find our documentation [here](https://skypilot.readthedocs.io/en/latest/
 
 ## Example SkyPilot Task
 
-A SkyPilot task is specified as a YAML file containing the resource requirements, data to be synced, setup commands, and the task commands. 
+A SkyPilot task is specified as a YAML file containing: resource requirements, data to be synced, setup commands, and the task commands. 
 
-Once written in a YAML, the task can be launched on any available cloud. Example:
+Once written in this [**unified interface**](https://skypilot.readthedocs.io/en/latest/reference/yaml-spec.html), the task can be launched on any available cloud. Example:
```
Comment (romilbhardwaj) [inline README.md:38]: Should we re-emphasize that this indeed a YAML, not some complex interface? e.g. `[**unified YAML interface**]`

**Review by romilbhardwaj — APPROVED** [review]
lgtm!

File: README.md (hunk 3 above)
Comment (Michaelvll) [reply README.md:38]: Maybe we can list both possible interface, like `Once written in this [**unified interface**](https://skypilot.readthedocs.io/en/latest/reference/yaml-spec.html) (YAML or Python API),`

File: README.md [hunk 4]
```diff
@@ -12,12 +12,12 @@ SkyPilot is a framework for easily running machine learning workloads[^1] on any
```
Comment (romilbhardwaj) [inline README.md:12]: Should we remove the word "easily" from L11? (since it's repeated again in L13)

Comment (concretevitamin) [reply README.md:12]: Good call, done.

File: README.md (hunk 1 above)
Comment (concretevitamin) [reply README.md:17]: Debated offline; keeping the current.

File: README.md (hunk 3 above)
Comment (concretevitamin) [reply README.md]: Good call, done.

File: README.md (hunk 2 above)
Comment (concretevitamin) [reply README.md]: Done.

**Review by concretevitamin — COMMENTED** [review]
Thanks, PTAL?

**Review by Michaelvll — APPROVED** [review]
LGTM! Thanks for the changes @concretevitamin!

**Pull request merged**